tempesta_fw/cache.c

```diff
@@ -523,14 +530,44 @@ tfw_cache_init(void)
 	return r;
 }
 
-void
-tfw_cache_exit(void)
+static void
+tfw_cache_stop(void)
 {
-	if (!tfw_cfg.cache)
+	if (!cache_cfg.cache)
 		return;
 
 	destroy_workqueue(cache_wq);
 	kmem_cache_destroy(c_cache);
 	kthread_stop(cache_mgr_thr);
-	tdb_close(db);
 }
+
+static TfwCfgSpec tfw_cache_cfg_specs[] = {
+	{
+		"cache", "off",
+		tfw_cfg_set_bool, &cache_cfg.cache
+	},
+	{
+		"cache_size", "262144",
+		tfw_cfg_set_int, &cache_cfg.db_size,
+		&(TfwCfgSpecInt) {
+			.is_multiple_of = PAGE_SIZE,
+			.range = { PAGE_SIZE, (1 << 30) },
+		}
+	},
+	{
+		"cache_path", "/opt/tempesta/cache",
+		tfw_cfg_set_str, &cache_cfg.db_path,
+		&(TfwCfgSpecStr) {
+			.len_range = { 0, PATH_MAX },
+			.buf = { (char[PATH_MAX]){}, PATH_MAX }
+		}
+	},
+	{}
+};
+
+TfwCfgMod tfw_mod_cache = {
+	.name 	 = "cache",
+	.start	 = tfw_cache_start,
+	.stop	 = tfw_cache_stop,
+	.specs	 = tfw_cache_cfg_specs,
```

Comment:
Please initialize structure members in the same order as they are declared - it makes the code easier to read.